docs/extensions/extension-registry.md

```diff
@@ -0,0 +1,580 @@
+
+```yaml
+id: mlflow
+product: mlflow
+version: 1.20.2
+description: |
+  MLFlow is an open source platform specialized in tracking ML experiments, and packaging and deploying ML models.
+services:
+- id: mlflow-tracking
+  resource: mlflow-tracking
+  category: experiment-tracking
+  authrequired: false
+  description: MLFlow experiment tracking service API and UI
+  endpoints:
+  - url: http://mlflow.mlflow
+    type: internal
+    configuration:
+      MLFLOW_TRACKING_URI: http://mlflow.mlflow
+- id: mlflow-store
+  resource: s3
+  category: model-store
+  authrequired: true
+  description: MLFlow minio S3 storage back-end
+  endpoints:
+  - url: http://mlflow-minio.mlflow:9000
+    type: internal
+    configuration:
+      MLFLOW_S3_ENDPOINT_URL: http://mlflow-minio.mlflow:9000
+  credentials:
+  - id: default-s3-account
+    default: false
+    scope: global
+    projects: []
+    users: []
+    configuration:
+      AWS_ACCESS_KEY_ID: <hidden>
+      AWS_SECRET_ACCESS_KEY: <hidden>
+```
+
+As can be seen from the descriptor, an MLFlow server consists of two services: the experiment tracking service and the artifact store. The experiment tracking service is responsible for tracking and visualizing ML experiments. The artifact store (in our case backed by minio - an S3 object store) is the service responsible for storing ML artifacts, such as ML models, datasets and logs. Services have individual endpoints through which they can be accessed. The artifact store in particular requires authentication, which is why credentials are also provided.
+
+!!! note
+
+    In different circumstances, we wouldn't need to include the artifact store back-end as a separate service in the extension descriptor, for example if the MLflow tracking server API would act as a central proxy intermediating uploading and downloading of artifacts. However, in this case, the external clients need to interact directly with the MLflow artifact store back-end to upload/download artifacts, which is why we need to cover it as a standalone service under the same extension.
+
+Next, let's take a closer look at the different sections (elements) that make up an Extension Record and explore their utility:
+
+```yaml
+id: mlflow
+product: mlflow
+version: 1.20.2
+description: |
+  MLFlow is an open source platform specialized in tracking ML experiments, and packaging and deploying ML models.
+```
+
+The root _extension_ model element represents a single instance or installation of a framework/platform/service/product developed and released or hosted under a unique name and operated as a single cohesive unit. Different installations of the same product can be grouped together based on the _product_ and _version_ they were installed from. They can also be grouped together based on the the infrastructure domain (i.e. location, region, zone, area or kubernetes cluster) where the extension is running, based on a _zone_ identifier.
```

Comment:
You mean _id_ element instead of _extension_, or don't you?

Reply:
I rephrased it a bit to be less confusing and to not refer to the root element as an _extension_ anymore.